http-middleware/cors.go

```diff
@@ -5,8 +5,26 @@ import (
 	"strings"
 )
 
-// CorsMiddleware adds Access-Control-Allow-Origin header to responses.
+// CorsMiddleware adds CORS headers to requests.
+// Shouldn't be use, instead use the combination of helper
+// functions below.
 func CorsMiddleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Access-Control-Allow-Origin", "*")
+		w.Header().Set("Access-Control-Allow-Methods", "PUT, POST, GET, DELETE")
+		w.Header().Set("Access-Control-Allow-Headers", "Authorization, Content-Type")
+
+		if r.Method == "OPTIONS" {
+			w.WriteHeader(http.StatusNoContent)
+			return
+		}
+
+		next.ServeHTTP(w, r)
+	})
+}
+
+// CorsMiddleware adds CORS Origin header to responses.
+func CorsMiddlewareV2(next http.Handler) http.Handler {
```

Comment:
How is this function different.  I thought that we should move it to v2 directory?

Reply:
It was a part of the v1.*.* version before we changed to v2

Reply:
https://github.com/SKF/go-utility/blob/v1.10.1/http-middleware/cors.go